### ISSUE TITLE
Crash when specifying seeds using different input formats: --num_seeds  [flag in command line] vs  "modelSeeds"  [JSON file option]

### ISSUE THREAD

**Issue by gjoses** [google-deepmind/alphafold3#622]
Hello,

We have encountered persistent issues running AF3 on Blackwell GPUs (NVIDIA RTX PRO 6000 Blackwell Max-Q Workstation Edition). After extensive troubleshooting—primarily guided by discussions in this GitHub issues page—we managed to install and run it via Docker and Singularity by carefully pinning specific dependency and CUDA sub-versions.

However, we observed a reproducible and unexpected behavior related to seed specification during inference:

When specifying `--num_seeds=N` (for any N between 2 and 100) as a command-line flag, the job crashes intermittently with a critical “GPU has fallen off the bus” error, forcing a full node reboot.

When specifying the same number of seeds in the input JSON file via `"modelSeeds": [1, 2, 3, 4, 5, ..., N],`
under the same environment, on the same node, and with an otherwise byte-identical job configuration, inference completes successfully with no errors.

We are currently working to systematically reproduce and document this behavior across different job sizes. Notably, this issue did not occur on pre-Blackwell GPUs.

Does this discrepancy align with any known behavior in AF3’s seed handling implementation, from the developers' point of view?

We hope this observation may be useful to others working with Blackwell hardware.

Best regards,
Gonzalo

**Comment by Augustin-Zidek** [comment]
Hello,

this sounds rather strange. `--num_seeds=N` just internally modifies the input JSON and fills in `N` seeds (from the given `rng_seed` to `rng_seed + N - 1`), so it should be equivalent.

I.e.

```json
"modelSeeds": [10]
```

launched with `--num_seeds=3` and

```json
"modelSeeds": [10, 11, 12]
```

should be equivalent.

Now that Blackwell GPUs are fully supported, would you mind syncing to the latest commit, rebuilding the container, and trying to run again? It is possible something got messed up when doing the hacky AF3 setup to enable B200 support.

**Comment by gjoses** [comment]
Thanks a lot @Augustin-Zidek for your response (and all the brilliant work with AF3!). We will certainly rebuild the container using the latest commit and will document everything. It might take a while.
Best regards,
Gonzalo

**Comment by arpittkhandelwal** [comment]
Hi sir,
I’ve resolved the issue and updated the PR accordingly. The CLA check is now passing as well.
Could you please take a look when you have time?
Thank you.

**Comment by arpittkhandelwal** [comment]
@Augustin-Zidek  Hi sir,
I’ve resolved the issue and updated the PR accordingly. The CLA check is now passing as well.
Could you please take a look when you have time?
Thank you.

**Comment by Augustin-Zidek** [comment]
Any luck, @gjoses?

**Comment by gjoses** [comment]
@Augustin-Zidek yes, we finally managed to consistently eliminate the `--num_seeds=N` (command line) vs.  `"modelSeeds": [1, 2, 3, 4, 5, ..., N]` (JSON file) issue with our NVIDIA RTX PRO 6000 Blackwell GPUs. This was achieved by rebuilding the container using the latest commit (as of 18 Feb 2026), and replacing `FROM nvidia/cuda:12.6.3-base-ubuntu24.04` with `FROM nvidia/cuda:13.1.1-base-ubuntu24.04` in `docker/Dockerfile`.
I am attaching `pyproject.toml`, `uv.lock` and `docker/Dockerfile` for reference.
Best performance (i.e. no issues) was achieved by installing CUDA 13.0 (`NVIDIA-SMI 580.126.09 Driver Version: 580.126.09 CUDA Version: 13.0`) in our computing node hosting four Blackwell GPUs. CUDA 13.1 was a bit less reliable, so we downgraded to CUDA 13.0 for production.

We used `Apptainer 1.4.5` to create and run production SIF images.

We are seeing the following warnings in our slurm.out files, which I assume are not problematic for typical runs:
```
I0221 21:50:49.187543 22669481915136 cache.py:95] Autotuning cache file not found: /alphafold3_venv/lib/python3.12/site-packages/tokamax/data/autotuning/nvidia_rtx_pro_6000_blackwell_max-q_workstation_edition/pallas_triton_gated_linear_unit.json
/app/alphafold/src/alphafold3/model/network/featurization.py:104: UserWarning: Explicitly requested dtype int64 requested in broadcasted_iota is not available, and will be truncated to dtype int32. To enable more dtypes, set the jax_enable_x64 configuration option or the JAX_ENABLE_X64 shell environment variable. See https://github.com/jax-ml/jax#current-gotchas for more.
  iota = jax.lax.broadcasted_iota(jnp.int64, logits.shape, axis)
```

Finally, since we typically run SLURM arrays consisting of hundreds of protein predictions with similar sizes and token counts, the flag `--jax_compilation_cache_dir=/root/af_output/jax_cache` has been found to be very helpful to reuse previously compiled kernels across runs, reducing compilation overhead and accelerating inference substantially.

I sincerely appreciate your patience and excellent work with AlphaFold3.

Best regards,
Gonzalo


[pyproject.toml.txt](https://github.com/user-attachments/files/25461482/pyproject.toml.txt)

[uv.lock.txt](https://github.com/user-attachments/files/25461497/uv.lock.txt)

[Dockerfile.txt](https://github.com/user-attachments/files/25461499/Dockerfile.txt)

**Comment by Augustin-Zidek** [comment]
Hi @gjoses, great to hear it now works for you! 

Thank you also for confirming and also describing what works -- this is very useful to hear as usually in GitHub issues I get only samples of things that _don't_ work, but you provided a sample of things what _do_ work!

> Best performance (i.e. no issues) was achieved by installing CUDA 13.0 

I am planning to upgrade the AF3 Docker to CUDA 13 soon.

> We are seeing the following warnings in our slurm.out files, which I assume are not problematic for typical runs:

I am tracking this in https://github.com/google-deepmind/alphafold3/issues/609 and a fix is coming on Tokamax side in https://github.com/openxla/tokamax/pull/660. These warnings are only about performance, not about correctness, so they are safe to ignore, but should be fixed soon and AF3 performance should improve slightly once we pull the updated Tokamax version.

> Finally, since we typically run SLURM arrays consisting of hundreds of protein predictions with similar sizes and token counts, the flag `--jax_compilation_cache_dir=/root/af_output/jax_cache` has been found to be very helpful to reuse previously compiled kernels across runs, reducing compilation overhead and accelerating inference substantially.

Great to hear this is working for you!
